al-khaser/AntiVM/Generic.cpp

```diff
@@ -96,7 +96,266 @@ VOID known_file_names() {
+	}
+
+	TCHAR msg[256];
+	WORD dwlength = sizeof(szUsernames) / sizeof(szUsernames[0]);
+	for (int i = 0; i < dwlength; i++) {
+
+		_stprintf_s(msg, sizeof(msg) / sizeof(msg[0]), _T("Checking if username matches : %s "), szUsernames[i]);
+
+		/* Do a case-insensitive search for all entries in szHostnames */
+		BOOL matched = FALSE;
+		if (0 == _tcsicmp(szUsernames[i], username)) {
+			matched = TRUE;
+		}
+
+		print_results(matched, msg);
+	}
+
+	free(username);
+}
+
+static TCHAR * get_netbios_hostname() {
+	TCHAR *hostname;
+	DWORD nSize = (MAX_COMPUTERNAME_LENGTH + 1);
+
+	hostname = (TCHAR *) malloc(nSize * sizeof(TCHAR));
+	if (!hostname) {
+		return NULL;
+	}
+	if (0 == GetComputerName(hostname, &nSize)) {
+		free(hostname);
+		return NULL;
+	}
+	return hostname;
+}
+
+static TCHAR * get_dns_hostname() {
+	TCHAR *hostname;
+	DWORD nSize = 0;
+
+	GetComputerNameEx(ComputerNameDnsHostname, NULL, &nSize);
+	hostname = (TCHAR *) malloc((nSize + 1) * sizeof(TCHAR));
+	if (!hostname) {
+		return NULL;
+	}
+	if (0 == GetComputerNameEx(ComputerNameDnsHostname, hostname, &nSize)) {
+		free(hostname);
+		return NULL;
+	}
+	return hostname;
+}
+
+/*
+Check for hostnames associated with sandboxes
+*/
+VOID known_hostnames() {
+
+	/* Array of strings of hostnames seen in sandboxes */
+	CONST TCHAR* szHostnames[] = {
+		/* Checked for by Gootkit
+		 * https://www.sentinelone.com/blog/gootkit-banking-trojan-deep-dive-anti-analysis-features/ */
+		_T("SANDBOX"),
+		_T("7SILVIA"),
+
+		/* Checked for by ostap
+		 * https://www.bromium.com/deobfuscating-ostap-trickbots-javascript-downloader/ */
+		_T("HANSPETER-PC"),
+		_T("JOHN-PC"),
+		_T("MUELLER-PC"),
+		_T("WIN7-TRAPS"),
+
+		/* Checked for by Shifu (not including ones from above)
+		 * https://www.mcafee.com/blogs/other-blogs/mcafee-labs/japanese-banking-trojan-shifu-combines-malware-tools */
+		_T("FORTINET"),
+
+		/* Checked for by Emotet (not including ones from above)
+		 * https://blog.trendmicro.com/trendlabs-security-intelligence/new-emotet-hijacks-windows-api-evades-sandbox-analysis/ */
+		_T("TEQUILABOOMBOOM"), /* VirusTotal Cuckoofork Sandbox */
+	};
+	TCHAR *NetBIOSHostName;
+	TCHAR *DNSHostName;
+
+	if (NULL == (NetBIOSHostName = get_netbios_hostname())) {
+		return;
+	}
+
+	if (NULL == (DNSHostName = get_dns_hostname())) {
+		free(NetBIOSHostName);
+		return;
+	}
+
+	TCHAR msg[256];
+	WORD dwlength = sizeof(szHostnames) / sizeof(szHostnames[0]);
+	for (int i = 0; i < dwlength; i++) {
+
+		_stprintf_s(msg, sizeof(msg) / sizeof(msg[0]), _T("Checking if hostname matches : %s "), szHostnames[i]);
+
+		/* Do a case-insensitive search for all entries in szHostnames */
+		BOOL matched = FALSE;
+		if (0 == _tcsicmp(szHostnames[i], NetBIOSHostName)) {
+			matched = TRUE;
+		}
+		else if (0 == _tcsicmp(szHostnames[i], DNSHostName)) {
+			matched = TRUE;
+		}
+
+		print_results(matched, msg);
+	}
+
+	free(NetBIOSHostName);
+	free(DNSHostName);
+}
+
+static BOOL check_file_exists(TCHAR *path) {
```

Comment:
There is a similar function in `Utils.cpp` called `is_FileExists()`. It uses `GetFileAttributes()` which can tells you that a file exists even if you don't have access to. Which is what you probably want here.

Reply:
Oh nice, thanks! I switched the code to use that helper function